crates/rspack_cacheable/src/context.rs

```diff
@@ -0,0 +1,65 @@
+use std::{any::Any, ptr::NonNull};
+
+use rkyv::{
+  de::{ErasedPtr, Pooling, PoolingState},
+  ser::{sharing::SharingState, Sharing},
+};
+
+use crate::{DeserializeError, SerializeError};
+
+const CONTEXT_ADDR: usize = 0;
+unsafe fn default_drop(_: ErasedPtr) {}
+
+/// A context wrapper that provides shared context methods
+pub struct ContextGuard<'a> {
+  context: &'a dyn Any,
+}
+
+impl<'a> ContextGuard<'a> {
+  pub fn new(context: &'a dyn Any) -> Self {
+    Self { context }
+  }
+
+  pub fn add_to_sharing<S: Sharing<SerializeError>>(
+    &self,
+    sharing: &mut S,
+  ) -> Result<(), SerializeError> {
+    sharing.start_sharing(CONTEXT_ADDR);
+    sharing.finish_sharing(CONTEXT_ADDR, self as *const _ as usize)
```

Comment:
Consider using [.addr](https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.addr) or [sptr](https://docs.rs/sptr/0.3.2/sptr/index.html) crate when cast pointers. but I understand it is not mandatory at the moment.